lib/_http_client.js

```diff
@@ -218,8 +222,11 @@ ClientRequest.prototype._finish = function _finish() {
 };
 
 ClientRequest.prototype._implicitHeader = function _implicitHeader() {
+  if (this._header) {
+    throw new Error('Can\'t render headers after they are sent to the client');
```

Comment:
It shouldn't, I just had to move the check from the old `_renderHeaders()` to the appropriate call sites since that function no longer exists. I tried just moving it to `_storeHeader()` but that caused problems IIRC.